### ISSUE TITLE
Resource Kind plugins

### ISSUE THREAD

**Issue by olensmar** [kubeshop/monokle#302]
I want to add support for custom resource kinds to monokle :
- see them in the Navigator under my defined section / subsection
- see their links to related resources
- have code completion and documentation popups in source editor
- be able to create new resources using the "New Resource" wizard
- be able to list/apply/diff rollout resources against my local cluster

This is dependent on #177

**Comment by olensmar** [comment]
great! can a git repo host more than one plugin?

**Comment by steenkamby** [comment]
A plugin is essentially a collection of plugin components, ui, terminal calls, etc. 

I think we should put a restriction of just one collection pr. repo to begin with maybe? Might make it simpler to parse and test? Our potential future plugin manager would have to be more complex if a plugin source has more than one plugin, in it.

**Comment by chlete** [comment]
Creating another ticket with more specific actions on the resource Kind
